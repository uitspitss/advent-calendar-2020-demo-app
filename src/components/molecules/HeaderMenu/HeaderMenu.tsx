import React, { FC } from 'react';
import tw, { styled } from 'twin.macro';
import Link from 'next/link';

export type HeaderMenuProps = {
  className?: string;
  children?: never;
};

type Props = HeaderMenuProps;

const Component: FC<Props> = (props) => {
  const { className } = props;

  return (
    <div className={className}>
      <Link href="/users-ssr">Users SSR</Link>
      <a href="/users-ssg">Users SSG</a>
      <a href="/users-swr">Users useSWR</a>
    </div>
  );
};

const StyledComponent = styled(Component)`
  ${tw`text-sm lg:flex-grow`}

  & > a {
    ${tw`block mt-4 lg:inline-block lg:mt-0 text-gray-500`}

    &:not(:last-child) {
      ${tw`mr-4`}
    }
  }
`;

/**
 * ヘッダー内のメニュー
 */
export const HeaderMenu: FC<HeaderMenuProps> = (props) => (
  <StyledComponent {...props} />
);
