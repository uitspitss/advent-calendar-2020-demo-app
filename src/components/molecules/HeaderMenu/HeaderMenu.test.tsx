import React from 'react';
import renderer from 'react-test-renderer';

import { HeaderMenu } from './HeaderMenu';

describe('HeaderMenuList', () => {
  test('Snap Shot', () => {
    const tree = renderer.create(<HeaderMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
