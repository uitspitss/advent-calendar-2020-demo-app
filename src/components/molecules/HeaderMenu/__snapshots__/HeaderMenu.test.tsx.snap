// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenuList Snap Shot 1`] = `
.emotion-0 {
  font-size: 0.875rem;
}

@media (min-width:1024px) {
  .emotion-0 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-0 > a {
  display: block;
  margin-top: 1rem;
  --text-opacity: 1;
  color: rgba(160,174,192,var(--text-opacity));
}

@media (min-width:1024px) {
  .emotion-0 > a {
    display: inline-block;
    margin-top: 0;
  }
}

.emotion-0 > a:not(:last-child) {
  margin-right: 1rem;
}

<div
  className="emotion-0 emotion-1"
>
  <a
    href="/"
  >
    Docs
  </a>
  <a
    href="/"
  >
    Examples
  </a>
  <a
    href="/"
  >
    Blog
  </a>
</div>
`;
